adb_client/src/usb/adb_usb_device.rs

```diff
@@ -37,6 +41,38 @@ fn read_adb_private_key<P: AsRef<Path>>(private_key_path: P) -> Result<Option<AD
             }
         })
 }
+/// Search for adb devices with known interface class and subclass values
+fn search_adb_devices() -> Option<(u16, u16)> {
+    for device in rusb::devices().unwrap().iter() {
+        let Ok(des) = device.device_descriptor() else {
+            continue;
+        };
+        if is_adb_device(&device, &des) {
+            return Some((des.vendor_id(), des.product_id()));
+        }
+    }
+    None
+}
+
+fn is_adb_device<T: UsbContext>(device: &Device<T>, des: &DeviceDescriptor) -> bool {
+    for n in 0..des.num_configurations() {
+        let Ok(config_des) = device.config_descriptor(n) else {
+            continue;
+        };
+        for interface in config_des.interfaces() {
+            for interface_des in interface.descriptors() {
+                let proto = interface_des.protocol_code();
+                let class = interface_des.class_code();
+                let subcl = interface_des.sub_class_code();
+                if proto == 1 && ((class == 0xff && subcl == 0x42) || (class == 0xdc && subcl == 2))
```

Comment:
Can you add a comment here explaining what these constants are ? And use them as const variables with a self explanatory name ?